lib/projects/localDev/LocalDevProcess.ts

```diff
@@ -0,0 +1,198 @@
+import { IntermediateRepresentationNodeLocalDev } from '@hubspot/project-parsing-lib/src/lib/types';
+import { translateForLocalDev } from '@hubspot/project-parsing-lib';
+import { Build } from '@hubspot/local-dev-lib/types/Build';
+import { Environment } from '@hubspot/local-dev-lib/types/Config';
+import path from 'path';
+
+import { ProjectConfig } from '../../../types/Projects';
+import { LocalDevState } from '../../../types/LocalDev';
+import LocalDevLogger from './LocalDevLogger';
+import DevServerManagerV2 from './DevServerManagerV2';
+import { EXIT_CODES } from '../../enums/exitCodes';
+import { mapToUserFriendlyName } from '@hubspot/project-parsing-lib/src/lib/transform';
+
+type LocalDevProcessConstructorOptions = {
+  targetProjectAccountId: number;
+  targetTestingAccountId: number;
+  projectConfig: ProjectConfig;
+  projectDir: string;
+  projectId: number;
+  debug?: boolean;
+  deployedBuild?: Build;
+  isGithubLinked: boolean;
+  initialProjectNodes: {
+    [key: string]: IntermediateRepresentationNodeLocalDev;
+  };
+  env: Environment;
+};
+
+class LocalDevProcess {
+  private state: LocalDevState;
+  private _logger: LocalDevLogger;
+  private devServerManager: DevServerManagerV2;
+  constructor({
+    targetProjectAccountId,
+    targetTestingAccountId,
+    projectConfig,
+    projectDir,
+    projectId,
+    debug,
+    deployedBuild,
+    isGithubLinked,
+    initialProjectNodes,
+    env,
+  }: LocalDevProcessConstructorOptions) {
+    this.state = {
```

Comment:
It could make sense for `LocalDevState` to be its own class in the future, especially if dev servers will be mutating it. For now, I think this object will suffice

Reply:
I think it might be a good idea to make the object immutable so that the other dev servers can't change it.  We would want to be the source of truth for this information and probably wouldn't want it to change.